services/provider/server/server_test.go

```diff
@@ -0,0 +1,242 @@
+
+	client := newFakeClient(t, object...)
+
+	server := &OCSProviderServer{
+		client:    client,
+		namespace: "openshift-storage",
+	}
+
+	consumerResource := &ocsv1alpha1.StorageConsumer{
+		Status: ocsv1alpha1.StorageConsumerStatus{
+			CephResources: []ocsv1alpha1.CephResourcesSpec{
+				{
+					Name: "cephblockpool",
+					Kind: "CephBlockPool",
+					CephClients: map[string]string{
+						"node":        "cephuser-rbd-node",
+						"provisioner": "cephuser-rbd-provisioner",
+					},
+				},
+				{
+					Name: "cephFilesystemSubVolumeGroup",
+					Kind: "CephFilesystemSubVolumeGroup",
+				},
+				{
+					Name: "cephuser-rbd-provisioner",
+					Kind: "CephClient",
+				},
+				{
+					Name: "cephuser-rbd-node",
+					Kind: "CephClient",
+				},
+				{
+					Name: "cephuser-cephfs-provisioner",
+					Kind: "CephClient",
+				},
+				{
+					Name: "cephuser-cephfs-node",
+					Kind: "CephClient",
+				},
+				{
+					Name: "cephuser-health-checker",
+					Kind: "CephClient",
+				},
+			},
+		},
+	}
+
+	for _, i := range consumerResource.Status.CephResources {
+		if i.Kind == "CephClient" {
+			cephClient := &rookCephv1.CephClient{
+				ObjectMeta: metav1.ObjectMeta{Name: i.Name, Namespace: server.namespace},
+				Status: &rookCephv1.CephClientStatus{
+					Info: map[string]string{
+						"secretName": fmt.Sprintf("rook-ceph-client-%s", i.Name),
+					},
+				},
+			}
+
+			assert.NoError(t, client.Create(ctx, cephClient))
+
+			name := cephClient.Status.Info["secretName"]
+
+			secret := &v1.Secret{
+				ObjectMeta: metav1.ObjectMeta{Name: name, Namespace: server.namespace},
+				Data: map[string][]byte{
+					i.Name: []byte("AQADw/hhqBOcORAAJY3fKIvte++L/zYhASjYPQ=="),
+				},
+			}
+			assert.NoError(t, client.Create(ctx, secret))
+
+		}
+
+	}
+
+	monCm := &v1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{Name: monConfigMap, Namespace: server.namespace},
+		Data:       map[string]string{"data": "a=10.99.45.27:6789", "mapping": "{}", "maxMonId": "0"},
+	}
+
+	assert.NoError(t, client.Create(ctx, monCm))
+
+	monSc := &v1.Secret{
+		ObjectMeta: metav1.ObjectMeta{Name: monSecret, Namespace: server.namespace},
+		Data: map[string][]byte{
+			"fsid": []byte("b88c2d78-9de9-4227-9313-a63f62f78743"),
+		},
+	}
+	assert.NoError(t, client.Create(ctx, monSc))
+
+	port, err := strconv.Atoi("9283")
+	assert.NoError(t, err)
+
+	mgrSvc := &v1.Service{
+		ObjectMeta: metav1.ObjectMeta{Name: "rook-ceph-mgr", Namespace: server.namespace},
+		Spec: v1.ServiceSpec{
+			ClusterIP: "10.105.164.231",
+			Ports: []v1.ServicePort{{
+				Port: int32(port),
+			}},
+		},
+	}
+	assert.NoError(t, client.Create(ctx, mgrSvc))
+
+	exR, err := server.getExternalResources(ctx, consumerResource)
+	assert.NoError(t, err)
+	for i := range exR {
+		data, err := json.Marshal(mockExtR[i].Data)
+		assert.NoError(t, err)
+		assert.Equal(t, string(exR[i].Data), string(data))
+		assert.Equal(t, exR[i].Name, mockExtR[i].Name)
+		assert.Equal(t, exR[i].Kind, mockExtR[i].Kind)
+
+	}
```

Comment:
This only checks the happy path scenario, what about testing failures as well?

Reply:
I'll update the unit test soon.